### PR TITLE
Nested rfactor fix, update predicates as well as source.

### PR DESCRIPTION
**Pull request by csarofeen**
Replace reduction axis in predicates as well as the source code. Fixes: https://github.com/dmlc/tvm/issues/3381

### CODE REVIEW

**Comment by tqchen** [comment]
Thanks for the contribution, @kazum @sgrechanik-h @derisavi @merrymercy please help to review this PR

File: src/schedule/schedule_dataflow_rewrite.cc [hunk 1]
```diff
@@ -756,12 +756,19 @@ Array<Tensor> Schedule::rfactor(const Tensor& tensor,
   VarReplacer replacer(vsub);
   Array<Expr> new_source = ir::UpdateArray(reduce->source,
     [&replacer] (const Expr& e) { return replacer.Mutate(e); });
+
+  Array<Expr> orig_pred;
+  orig_pred.push_back(predicate);
+
+  Array<Expr> new_pred = ir::UpdateArray(orig_pred,
+    [&replacer] (const Expr& e) { return replacer.Mutate(e); });
+
```
Comment (merrymercy) [inline src/schedule/schedule_dataflow_rewrite.cc]: Why do you create an array?
Just use
`Expr new_pred = replacer.Mutate(predicate)`

Comment (csarofeen) [reply src/schedule/schedule_dataflow_rewrite.cc]: Thank you for the syntax, updated.

**Comment by csarofeen** [comment]
@merrymercy Are there any other changes you'd like to see? https://github.com/dmlc/tvm/pull/3444 is also similar if you would please review that one as well, simply wraps the predicate in `likely(simplify(...))`.

**Pull request merged**